datastore/gcloud/aio/datastore/options.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+
+class ReadWrite:
+
+    def __init__(self, previous_transaction: str):
+        self.previous_transaction = previous_transaction
+
+    def to_repr(self) -> Dict[str, str]:
+        return {'previousTransaction': self.previous_transaction}
+
+
+class TransactionOptions:
+    def __init__(
+            self,
+            read_write: Optional[ReadWrite] = None,
+            read_only: Any = None):
+        self.read_write = read_write
+        self.read_only = read_only  # Type not documented
```

Comment:
Given the following comment in the docs:
> Union field mode can be only one of the following:

I suspect we'll want to make this an `enum.Enum` to avoid needing to handle the neither/both error cases.

I'm thinking something roughly along the lines of this:
```python
class ReadOnly:
    def to_repr(self) -> Dict[str, str]:
        return {}

class ReadWrite:
    def __init__(self, previous_transaction: str):
        self.previous_transaction = previous_transaction

    def to_repr(self) -> Dict[str, str]:
        return {'previousTransaction': self.previous_transaction}

class TransactionOptions:
    def __init__(self, option: Union[ReadOnly, ReadWrite]) -> None:
        self._option = option

    def to_repr(self) -> Dict[str, Dict[str, str]]:
        if isinstance(self.option, ReadOnly):
            return {'readOnly': self.option.to_repr()}
        if isinstance(self.option, ReadWrite):
            return {'readWrite': self.option.to_repr()}
        raise ValueError(f'invalid TransactionOptions {self.option}')
```

Thoughts?

---

Note that in other cases where the docs do not show a JSON representation, that has meant Google requires an empty dictionary as input. That said, it would be fantastic if we could test that assumption!

Reply:
Just as a counter point, it would be little bit confusing when there is some nesting. For example if you were to combine it with `ReadOptions` (which is not exposed right now) and you want a new transaction it would be:

```python
ReadOptions(TransactionOptions(ReadWrite()))
```  

instead of something that could be more clear:

```python
ReadOptions(new_transaction=TransactionOptions(read_write=ReadWrite()))
```

anyway it needs a validation that you don't provide both of them.